lib/api/api.go

```diff
@@ -258,6 +248,15 @@ func (s *service) Serve(ctx context.Context) error {
 
 	restMux := httprouter.New()
 
+	webauthnDeviceName := ""
+	if dev, ok := s.cfg.Device(s.id); ok {
```

Comment:
The device name is passed to  the webauthn engine - I assume it's of importance? Because currently if it changes, that change wont propagate. The api service only restarts on gui config changes.

Reply:
Ok, now followup with some actual pointers instead of just pointing out an issue (too tired before): The easiest option would be to just add a check to `api.service.CommitConfiguration` to also restart on device name change. Another option would be to add a mechanism to propagate a device name change from that function to the webauthn service. Or make the webauthnservice "standalone", i.e. let it subscribe to config changes instead of tagging on `api.service` for that (potentially make it even fully independent from api.service, i.e. it's own suture service, to also avoid the restarts - then again restarts when gui config changes might be desirable?).

Reply:
Nah, it's not very important at all - it's only used to set the [RP display name](https://www.w3.org/TR/2021/REC-webauthn-2-20210408/#dom-publickeycredentialentity-name) when registering credentials. That value may be presented in account picker or credential management UIs, but it's not used for any security decisions.

Actually, it's probably better anyway to just use the device ID rather than the name, because the name can change but the ID is immutable, and as of now there is no way for the RP (Syncthing) to signal to authenticators that the RP's display name should be updated (one might be [coming in WebAuthn L3](https://github.com/w3c/webauthn/pull/2093), but there are authenticators that will never support this anyway). So I think I'll just do that instead and eliminate the problem altogether.